### PR TITLE
docs: add jsdoc for error, body and some utils

### PR DESCRIPTION
**Pull request by farnabaz**

### CODE REVIEW

**Comment by Atinux** [comment]
Thanks @farnabaz 

While we are at this, I would like to suggest to rename `useBody` to `useBodyRaw` and having the parsing with `destr` by default for `useBody`.

Do you agree @pi0 ?

File: README.md [hunk 1]
```diff
@@ -103,6 +103,74 @@ When calling `app.use`, middleware will be automatically promisified.
...
+`h2` provides multiple utility functions.
+
+### `useQuery`
+
+Parse query parametes of the request and returns equivalent object.
+
+```js
+import { useQuery } from '@nuxt/h2';
+
+app.use('/api', (req) => ({ queryParams: useQuery(req) }))
```
Comment (Atinux) [inline README.md:117]:
```suggestion
app.use('/api', (req) => ({ query: useQuery(req) }))
```

I will also show a quick url example:

/api?hello=world&array[]=one&array=two will return `{ hello: 'world', array: ['one', 'two] }`

**Review by pi0 — CHANGES_REQUESTED** [review]
This is partial documentation which might be misleading. We shall include docs for all utils (like including sendError, promisify, etc)  or not. As suggested in other PR we may use JSDocs to maintain update docs with utils

**Comment by Atinux** [comment]
I see, fully agree with JSDocs for automated docs 🔥

**Comment by pi0** [comment]
> While we are at this, I would like to suggest to rename useBody to useBodyRaw and having the parsing with destr by default for useBody.

Makes sense

**Comment by farnabaz** [comment]
I've started using jsdoc and added comments to some functions. It is not completed yet.  
The result is available here: http://h2-docs.surge.sh/

**Comment by pi0** [comment]
@farnabaz Kindly please stop doing actions without consulting repo maintainer via issues!! This time I revert unnecessary changes but next time will simply close PR. Thanks.

**Pull request merged**